### PR TITLE
Enhancement/#47 oop

### LINKED ISSUE #47: Refactor for better OOP
Refactor code base to facilitate classes.
This makes for a more testable source code.

### PR DESCRIPTION
**Pull request by Crevil**
closes #47 

# Changes
* Moved things around to match the architecture layers.
* Added classes to functionality where applicable.
* Updated optics typings to the one in npm, until https://github.com/DefinitelyTyped/DefinitelyTyped/pull/13457 is merged.

### CODE REVIEW

File: optics.d.ts [hunk 1]
```diff
@@ -1,17 +1,83 @@
...
+export function context(req: Request): any;
+
+export default {
+  configureAgent,
+  instrumentSchema,
+  middleware,
+  instrumentHapiServer,
+  context,
+  Agent: OpticsAgent.Agent,
+};
```
Comment (Karnich) [inline optics.d.ts:82]: shouldnt this file be removed if your using the one from npm?

File: src/services/algorithms/analysisAlgorithm.ts [hunk 2]
```diff
@@ -221,8 +221,8 @@ export class AnalysisAlgorithm {
               Math.sqrt(
                 Math.pow(gyrXSeq[j][antalSamples - 1], 2) +
                 Math.pow(gyrYSeq[j][antalSamples - 1], 2) +
-                Math.pow(gyrZSeq[j][antalSamples - 1], 2)
-              )
+                Math.pow(gyrZSeq[j][antalSamples - 1], 2),
```
Comment (Karnich) [inline src/services/algorithms/analysisAlgorithm.ts]: is that , correct?

File: optics.d.ts (hunk 1 above)
Comment (Crevil) [reply optics.d.ts:82]: It is not published yet, so I just updated this file with the same code as the NPM module.

File: src/services/algorithms/analysisAlgorithm.ts (hunk 2 above)
Comment (Crevil) [reply src/services/algorithms/analysisAlgorithm.ts]: Good catch. Dunno why this have been added.

**Pull request merged**